spec/image_spec.rb

```diff
@@ -26,200 +24,248 @@
+        expect(assembly_image.exif.samplesperpixel).to be 3
+        expect(assembly_image.exif.bitspersample).to eql '8 8 8'
+        expect(assembly_image).to be_a_valid_image
+        expect(assembly_image).to be_jp2able
         expect(assembly_image).not_to have_color_profile
-        result = assembly_image.create_jp2(output: TEST_JP2_OUTPUT_FILE)
-        expect(TEST_JP2_OUTPUT_FILE).to be_a_jp2
+        result = assembly_image.create_jp2(output: output_file)
+        expect(result).to be_a_kind_of described_class
+        expect(result.path).to eq output_file
+        expect(output_file).to be_a_jp2
         expect(result.exif.colorspace).to eq 'sRGB'
       end
     end
 
+    context 'when given a cmyk tif' do
+      let(:input_path) { File.join(TEST_INPUT_DIR, 'cmky.tif') }
+
+      before do
+        generate_test_image(input_path, color: 'cmyk', cg_type: 'cmyk', profile: 'cmyk', bands: 4)
+      end
+
+      it 'creates jp2' do
+        expect(File).to exist input_path
+        expect(File).not_to exist output_file
+        expect(assembly_image.exif.samplesperpixel).to be 4
+        expect(assembly_image.exif.bitspersample).to eql '8 8 8 8'
+        expect(assembly_image).to be_a_valid_image
+        expect(assembly_image).to be_jp2able
+        expect(assembly_image).to have_color_profile
+        result = assembly_image.create_jp2(output: output_file)
+        expect(result).to be_a_kind_of described_class
+        expect(result.path).to eq output_file
+        expect(output_file).to be_a_jp2
+        # TODO: figure out why result.exif.colorspace is nil
+        # expect(result.exif.colorspace).to eq 'sRGB'
+      end
+    end
+
     context 'when the source image has no profile' do
+      let(:input_path) { File.join(TEST_INPUT_DIR, 'no_profile.tif') }
+
       before do
-        generate_test_image(TEST_TIF_INPUT_FILE, profile: '') # generate a test input with no profile
+        generate_test_image(input_path)
       end
 
       it 'creates a jp2' do
-        expect(File).to exist TEST_TIF_INPUT_FILE
-        expect(File).not_to exist TEST_JP2_OUTPUT_FILE
+        expect(File).to exist input_path
+        expect(File).not_to exist output_file
+        expect(assembly_image.exif.samplesperpixel).to be 3
+        expect(assembly_image.exif.bitspersample).to eql '8 8 8'
         expect(assembly_image).not_to have_color_profile
         expect(assembly_image).to be_a_valid_image
         expect(assembly_image).to be_jp2able
-        assembly_image.create_jp2(output: TEST_JP2_OUTPUT_FILE)
-        expect(TEST_JP2_OUTPUT_FILE).to be_a_jp2
+        assembly_image.create_jp2(output: output_file)
+        expect(output_file).to be_a_jp2
       end
     end
 
     context "when the output file exists and you don't allow overwriting" do
       before do
-        generate_test_image(TEST_TIF_INPUT_FILE)
-        generate_test_image(TEST_JP2_OUTPUT_FILE)
+        generate_test_image(input_path)
```

Comment:
It's a little hinky to have to scroll around in tests to find out what "input path" is.

Reply:
the idea behind this is that in some cases we want to the `input_path` to be unique for the test to be sure we get a different filename coming in, so we set it with a let statement in the specific test block, and then use whatever the value of `input_path` is to instantiate the class for all of the tests in the same way (in line 7)